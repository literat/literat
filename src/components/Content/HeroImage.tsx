import React from 'react';
import styled from 'styled-components';
import Img from '../Img';

const HeroImageStyles = styled.div`
  * {
    width: 140%;
    margin-inline: -20%;
  }
`;
function HeroImage({ image, title }) {
  console.log(image);
  return (
    <HeroImageStyles>
      <Img image={image} alt={title} />
    </HeroImageStyles>
  );
}

export default HeroImage;
