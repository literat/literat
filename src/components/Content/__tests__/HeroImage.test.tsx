import React from 'react';
import { render, screen } from '@testing-library/react';
import HeroImage from '../HeroImage';

describe('HeroImage', () => {
  it('should render', () => {
    const image = {
      childImageSharp: {
        gatsbyImageData: {
          images: {
            fallback: {
              src: 'https://via.placeholder.com/900x600',
            },
          },
        },
      },
    };
    const title = 'test title';
    const dom = render(<HeroImage image={image} title={title} />);

    expect(dom.container.querySelector('img')).toMatchSnapshot();
  });
});
