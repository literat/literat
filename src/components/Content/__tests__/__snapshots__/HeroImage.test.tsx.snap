// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage should render 1`] = `
<img
  alt="test title"
  data-gatsby-image-ssr=""
  data-main-image=""
  data-src="https://via.placeholder.com/900x600"
  decoding="async"
  loading="lazy"
  style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
/>
`;
